### PR TITLE
Stage 1

### PR DESCRIPTION
**Pull request by KateMoreva**

### CODE REVIEW

**Comment by KateMoreva** [comment]
Тест 1
Снимок экрана от 2020-09-27 13-50-03
Снимок экрана от 2020-09-27 13-50-20

**Comment by KateMoreva** [comment]
Тест 1
Снимок экрана от 2020-09-27 17-04-57
Снимок экрана от 2020-09-27 17-04-26

**Comment by KateMoreva** [comment]
Тест2
Снимок экрана от 2020-09-27 18-29-41
Снимок экрана от 2020-09-27 18-29-11

**Comment by KateMoreva** [comment]
Тест3
Снимок экрана от 2020-09-27 18-33-54
Снимок экрана от 2020-09-27 18-33-13

**Comment by KateMoreva** [comment]
sudo wrk -t1 -c1 -d30s -R20000 --latency http://localhost:8080/v0/status


Running 30s test @ http://localhost:8080/v0/status
  1 threads and 1 connections
  Thread calibration: mean lat.: 7.054ms, rate sampling interval: 34ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     7.42ms    7.23ms  42.40ms   82.59%
    Req/Sec    20.29k     4.97k   41.31k    70.02%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%    5.42ms
 75.000%   10.62ms
 90.000%   17.49ms
 99.000%   32.42ms
 99.900%   40.54ms
 99.990%   42.30ms
 99.999%   42.40ms
100.000%   42.43ms

  Detailed Percentile spectrum:
       Value   Percentile   TotalCount 1/(1-Percentile)

       0.014     0.000000            2         1.00
       0.653     0.100000        40002         1.11
       1.205     0.200000        79977         1.25
       2.289     0.300000       119934         1.43
       3.815     0.400000       159918         1.67
       5.423     0.500000       199877         2.00
       6.271     0.550000       219908         2.22
       7.131     0.600000       239843         2.50
       8.115     0.650000       259825         2.86
       9.263     0.700000       279806         3.33
      10.623     0.750000       299798         4.00
      11.383     0.775000       309846         4.44
      12.255     0.800000       319777         5.00
      13.295     0.825000       329768         5.71
      14.567     0.850000       339759         6.67
      15.935     0.875000       349755         8.00
      16.671     0.887500       354777         8.89
      17.487     0.900000       359773        10.00
      18.463     0.912500       364811        11.43
      19.487     0.925000       369748        13.33
      20.639     0.937500       374774        16.00
      21.263     0.943750       377239        17.78
      21.823     0.950000       379762        20.00
      22.783     0.956250       382219        22.86
      24.063     0.962500       384725        26.67
      25.391     0.968750       387212        32.00
      26.351     0.971875       388459        35.56
      27.343     0.975000       389706        40.00
      28.351     0.978125       390968        45.71
      29.215     0.981250       392222        53.33
      30.111     0.984375       393453        64.00
      30.575     0.985938       394081        71.11
      31.183     0.987500       394708        80.00
      32.063     0.989062       395333        91.43
      32.735     0.990625       395962       106.67
      33.599     0.992188       396583       128.00
      34.015     0.992969       396889       142.22
      34.463     0.993750       397203       160.00
      34.847     0.994531       397521       182.86
      35.263     0.995313       397835       213.33
      35.807     0.996094       398137       256.00
      36.511     0.996484       398295       284.44
      37.215     0.996875       398452       320.00
      37.919     0.997266       398612       365.71
      38.559     0.997656       398764       426.67
      39.199     0.998047       398924       512.00
      39.519     0.998242       399002       568.89
      39.839     0.998437       399083       640.00
      40.127     0.998633       399154       731.43
      40.351     0.998828       399230       853.33
      40.543     0.999023       399314      1024.00
      40.639     0.999121       399356      1137.78
      40.735     0.999219       399387      1280.00
      41.023     0.999316       399426      1462.86
      41.311     0.999414       399464      1706.67
      41.599     0.999512       399505      2048.00
      41.759     0.999561       399527      2275.56
      41.887     0.999609       399544      2560.00
      42.015     0.999658       399562      2925.71
      42.111     0.999707       399584      3413.33
      42.175     0.999756       399605      4096.00
      42.207     0.999780       399616      4551.11
      42.239     0.999805       399631      5120.00
      42.239     0.999829       399631      5851.43
      42.271     0.999854       399642      6826.67
      42.303     0.999878       399661      8192.00
      42.303     0.999890       399661      9102.22
      42.303     0.999902       399661     10240.00
      42.335     0.999915       399672     11702.86
      42.335     0.999927       399672     13653.33
      42.367     0.999939       399682     16384.00
      42.367     0.999945       399682     18204.44
      42.367     0.999951       399682     20480.00
      42.367     0.999957       399682     23405.71
      42.399     0.999963       399697     27306.67
      42.399     0.999969       399697     32768.00
      42.399     0.999973       399697     36408.89
      42.399     0.999976       399697     40960.00
      42.399     0.999979       399697     46811.43
      42.399     0.999982       399697     54613.33
      42.399     0.999985       399697     65536.00
      42.399     0.999986       399697     72817.78
      42.399     0.999988       399697     81920.00
      42.399     0.999989       399697     93622.86
      42.399     0.999991       399697    109226.67
      42.399     0.999992       399697    131072.00
      42.399     0.999993       399697    145635.56
      42.399     0.999994       399697    163840.00
      42.399     0.999995       399697    187245.71
      42.399     0.999995       399697    218453.33
      42.399     0.999996       399697    262144.00
      42.399     0.999997       399697    291271.11
      42.399     0.999997       399697    327680.00
      42.399     0.999997       399697    374491.43
      42.431     0.999998       399698    436906.67
      42.431     1.000000       399698          inf
Mean    =        7.422, StdDeviation   =        7.232
Max     =       42.400, Total count    =       399698
Buckets =           27, SubBuckets     =         2048
----------------------------------------------------------
  599699 requests in 30.00s, 35.46MB read
Requests/sec:  19989.99
Transfer/sec:      1.18MB

**Comment by KateMoreva** [comment]
sudo wrk -t1 -c1 -d60s -R20000 -s ./wrk/put.lua --latency http://localhost:8080

Running 1m test @ http://localhost:8080
  1 threads and 1 connections
  Thread calibration: mean lat.: 7.821ms, rate sampling interval: 43ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     7.39ms    9.08ms  58.24ms   85.68%
    Req/Sec    20.24k     4.93k   42.90k    73.32%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%    3.39ms
 75.000%   10.08ms
 90.000%   20.56ms
 99.000%   41.06ms
 99.900%   53.73ms
 99.990%   57.66ms
 99.999%   58.24ms
100.000%   58.27ms

  Detailed Percentile spectrum:
       Value   Percentile   TotalCount 1/(1-Percentile)

       0.019     0.000000            2         1.00
       0.485     0.100000       100098         1.11
       0.909     0.200000       200089         1.25
       1.531     0.300000       300060         1.43
       2.647     0.400000       400103         1.67
       3.389     0.500000       500127         2.00
       4.303     0.550000       550099         2.22
       5.491     0.600000       600111         2.50
       6.783     0.650000       650050         2.86
       8.207     0.700000       700106         3.33
      10.079     0.750000       750092         4.00
      11.183     0.775000       775042         4.44
      12.471     0.800000       800098         5.00
      13.991     0.825000       825039         5.71
      15.975     0.850000       850062         6.67
      18.095     0.875000       875061         8.00
      19.295     0.887500       887605         8.89
      20.559     0.900000       900011        10.00
      22.015     0.912500       912631        11.43
      23.423     0.925000       925082        13.33
      25.023     0.937500       937509        16.00
      26.031     0.943750       943829        17.78
      26.895     0.950000       950125        20.00
      28.031     0.956250       956316        22.86
      29.519     0.962500       962543        26.67
      31.343     0.968750       968786        32.00
      32.175     0.971875       971981        35.56
      33.055     0.975000       975071        40.00
      33.951     0.978125       978168        45.71
      34.975     0.981250       981274        53.33
      36.799     0.984375       984382        64.00
      37.727     0.985938       985976        71.11
      38.751     0.987500       987509        80.00
      40.223     0.989062       989081        91.43
      41.599     0.990625       990630       106.67
      43.103     0.992188       992214       128.00
      43.711     0.992969       992987       142.22
      44.703     0.993750       993766       160.00
      45.695     0.994531       994549       182.86
      46.687     0.995313       995322       213.33
      47.903     0.996094       996101       256.00
      48.383     0.996484       996492       284.44
      48.831     0.996875       996884       320.00
      49.599     0.997266       997275       365.71
      50.303     0.997656       997666       426.67
      51.455     0.998047       998062       512.00
      51.967     0.998242       998249       568.89
      52.543     0.998437       998449       640.00
      53.055     0.998633       998649       731.43
      53.407     0.998828       998838       853.33
      53.759     0.999023       999031      1024.00
      53.919     0.999121       999142      1137.78
      54.079     0.999219       999241      1280.00
      54.143     0.999316       999405      1462.86
      54.175     0.999414       999444      1706.67
      54.271     0.999512       999519      2048.00
      54.367     0.999561       999571      2275.56
      54.463     0.999609       999616      2560.00
      55.199     0.999658       999664      2925.71
      55.775     0.999707       999714      3413.33
      56.351     0.999756       999763      4096.00
      56.575     0.999780       999786      4551.11
      56.799     0.999805       999811      5120.00
      57.023     0.999829       999838      5851.43
      57.247     0.999854       999863      6826.67
      57.471     0.999878       999884      8192.00
      57.567     0.999890       999896      9102.22
      57.695     0.999902       999910     10240.00
      57.791     0.999915       999923     11702.86
      57.855     0.999927       999932     13653.33
      57.951     0.999939       999946     16384.00
      57.983     0.999945       999951     18204.44
      58.047     0.999951       999962     20480.00
      58.079     0.999957       999968     23405.71
      58.111     0.999963       999974     27306.67
      58.143     0.999969       999980     32768.00
      58.143     0.999973       999980     36408.89
      58.175     0.999976       999987     40960.00
      58.175     0.999979       999987     46811.43
      58.175     0.999982       999987     54613.33
      58.207     0.999985       999994     65536.00
      58.207     0.999986       999994     72817.78
      58.207     0.999988       999994     81920.00
      58.239     0.999989      1000004     93622.86
      58.239     0.999991      1000004    109226.67
      58.239     0.999992      1000004    131072.00
      58.239     0.999993      1000004    145635.56
      58.239     0.999994      1000004    163840.00
      58.239     0.999995      1000004    187245.71
      58.239     0.999995      1000004    218453.33
      58.239     0.999996      1000004    262144.00
      58.239     0.999997      1000004    291271.11
      58.239     0.999997      1000004    327680.00
      58.239     0.999997      1000004    374491.43
      58.239     0.999998      1000004    436906.67
      58.239     0.999998      1000004    524288.00
      58.239     0.999998      1000004    582542.22
      58.239     0.999998      1000004    655360.00
      58.239     0.999999      1000004    748982.86
      58.239     0.999999      1000004    873813.33
      58.271     0.999999      1000005   1048576.00
      58.271     1.000000      1000005          inf
Mean    =        7.393, StdDeviation   =        9.080
Max     =       58.240, Total count    =      1000005
Buckets =           27, SubBuckets     =         2048
----------------------------------------------------------
  1199985 requests in 1.00m, 81.25MB read
  Non-2xx or 3xx responses: 1199985
Requests/sec:  19999.77
Transfer/sec:      1.35MB

**Comment by KateMoreva** [comment]
Running 30s test @ http://localhost:8080
  1 threads and 1 connections
  Thread calibration: mean lat.: 6.991ms, rate sampling interval: 36ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     5.70ms    6.38ms  44.77ms   85.89%
    Req/Sec    20.30k     4.60k   41.14k    70.63%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%    3.11ms
 75.000%    8.17ms
 90.000%   14.76ms
 99.000%   26.88ms
 99.900%   42.53ms
 99.990%   44.67ms
 99.999%   44.77ms
100.000%   44.80ms

  Detailed Percentile spectrum:
       Value   Percentile   TotalCount 1/(1-Percentile)

       0.018     0.000000            1         1.00
       0.479     0.100000        40082         1.11
       0.884     0.200000        80114         1.25
       1.401     0.300000       120108         1.43
       2.379     0.400000       160174         1.67
       3.113     0.500000       200292         2.00
       3.639     0.550000       220252         2.22
       4.615     0.600000       240310         2.50
       5.747     0.650000       260256         2.86
       6.839     0.700000       280260         3.33
       8.171     0.750000       300270         4.00
       8.879     0.775000       310297         4.44
       9.647     0.800000       320391         5.00
      10.575     0.825000       330330         5.71
      11.671     0.850000       340345         6.67
      12.975     0.875000       350328         8.00
      13.807     0.887500       355327         8.89
      14.759     0.900000       360355        10.00
      15.727     0.912500       365339        11.43
      17.039     0.925000       370361        13.33
      18.351     0.937500       375382        16.00
      19.071     0.943750       377876        17.78
      19.807     0.950000       380356        20.00
      20.543     0.956250       382845        22.86
      21.375     0.962500       385371        26.67
      22.191     0.968750       387855        32.00
      22.591     0.971875       389129        35.56
      23.135     0.975000       390348        40.00
      23.663     0.978125       391596        45.71
      24.399     0.981250       392861        53.33
      25.103     0.984375       394116        64.00
      25.487     0.985938       394733        71.11
      25.855     0.987500       395352        80.00
      26.479     0.989062       395975        91.43
      27.055     0.990625       396628       106.67
      27.631     0.992188       397226       128.00
      27.951     0.992969       397551       142.22
      28.383     0.993750       397862       160.00
      28.879     0.994531       398178       182.86
      30.639     0.995313       398480       213.33
      32.287     0.996094       398790       256.00
      32.799     0.996484       398957       284.44
      33.695     0.996875       399102       320.00
      35.551     0.997266       399260       365.71
      37.183     0.997656       399416       426.67
      38.879     0.998047       399572       512.00
      39.615     0.998242       399652       568.89
      40.415     0.998437       399729       640.00
      41.247     0.998633       399806       731.43
      41.919     0.998828       399884       853.33
      42.591     0.999023       399963      1024.00
      42.943     0.999121       400005      1137.78
      43.295     0.999219       400043      1280.00
      43.647     0.999316       400081      1462.86
      43.903     0.999414       400121      1706.67
      44.127     0.999512       400163      2048.00
      44.223     0.999561       400182      2275.56
      44.319     0.999609       400199      2560.00
      44.447     0.999658       400223      2925.71
      44.511     0.999707       400256      3413.33
      44.511     0.999756       400256      4096.00
      44.575     0.999780       400288      4551.11
      44.575     0.999805       400288      5120.00
      44.575     0.999829       400288      5851.43
      44.607     0.999854       400299      6826.67
      44.639     0.999878       400307      8192.00
      44.671     0.999890       400319      9102.22
      44.671     0.999902       400319     10240.00
      44.671     0.999915       400319     11702.86
      44.703     0.999927       400329     13653.33
      44.703     0.999939       400329     16384.00
      44.735     0.999945       400339     18204.44
      44.735     0.999951       400339     20480.00
      44.735     0.999957       400339     23405.71
      44.735     0.999963       400339     27306.67
      44.767     0.999969       400349     32768.00
      44.767     0.999973       400349     36408.89
      44.767     0.999976       400349     40960.00
      44.767     0.999979       400349     46811.43
      44.767     0.999982       400349     54613.33
      44.767     0.999985       400349     65536.00
      44.767     0.999986       400349     72817.78
      44.767     0.999988       400349     81920.00
      44.767     0.999989       400349     93622.86
      44.799     0.999991       400353    109226.67
      44.799     1.000000       400353          inf
Mean    =        5.696, StdDeviation   =        6.380
Max     =       44.768, Total count    =       400353
Buckets =           27, SubBuckets     =         2048
----------------------------------------------------------
  599970 requests in 30.00s, 40.62MB read
  Non-2xx or 3xx responses: 599970
Requests/sec:  19999.05
Transfer/sec:      1.35MB

**Comment by KateMoreva** [comment]
Анализ результатов тестирования будет добавлен сегодня

File: src/main/java/ru/mail/polis/service/kate/moreva/MySimpleHttpServer.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+     *         {@code 500} (internal server error occurred).
+     * */
+    private Response getEntity(final ByteBuffer key) {
+        try {
+            final ByteBuffer value = dao.get(key).duplicate();
+            final byte[] body = new byte[value.remaining()];
+            value.get(body);
+            return new Response(Response.OK, body);
+        } catch (NoSuchElementException e) {
+            return new Response(Response.NOT_FOUND, Response.EMPTY);
```
Comment (vaddya) [inline src/main/java/ru/mail/polis/service/kate/moreva/MySimpleHttpServer.java:103]: Может подтянем какую-нибудь либу для логирования и залогируем? Пригодиться в будущем

File: src/main/java/ru/mail/polis/service/kate/moreva/MySimpleHttpServer.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        } catch (IOException e) {
+            return new Response(Response.INTERNAL_ERROR, Response.EMPTY);
```
Comment (vaddya) [inline src/main/java/ru/mail/polis/service/kate/moreva/MySimpleHttpServer.java:105]: Может подтянем какую-нибудь либу для логирования и залогируем? Пригодится в будущем

**Review by vaddya — CHANGES_REQUESTED** [review]
Если посмотреть в флейм графы, то видно, что все запросы обрабатываются в `ru/mail/polis/service/kate/moreva/MySimpleHttpServer.handleDefault`, то есть мы никогда не попадаем в метод `MySimpleHttpServer.entity`. Предположу, что при обстреле использовался неверный URL.

`Non-2xx or 3xx responses: 1199985` означает, что что-то идет не так. Наверное, решится после фикса предыдущей проблемы.

Проводить профилирование нужно не толькопо `cpu` и `alloc`, но и по `latency`. Анализ (вместе с результатами `wrk`) будет лучше закоммитить в `.md` файлике. Круто, что кроме самих файлов, есть еще и их анализ. Предлагаю немного доработать до 10 баллов.

**Comment by KateMoreva** [comment]
> Если посмотреть в флейм графы, то видно, что все запросы обрабатываются в `ru/mail/polis/service/kate/moreva/MySimpleHttpServer.handleDefault`, то есть мы никогда не попадаем в метод `MySimpleHttpServer.entity`. Предположу, что при обстреле использовался неверный URL.
> 
> `Non-2xx or 3xx responses: 1199985` означает, что что-то идет не так. Наверное, решится после фикса предыдущей проблемы.
> 
> Проводить профилирование нужно не толькопо `cpu` и `alloc`, но и по `latency`. Анализ (вместе с результатами `wrk`) будет лучше закоммитить в `.md` файлике. Круто, что кроме самих файлов, есть еще и их анализ. Предлагаю немного доработать до 10 баллов.

Спасибо, действительно была ошибка в скрипте.
Провела заново все тесты.
Иходники с анализом лежат в папке tests.
Сейчас добавлю краткую сводку сюда для удобства.

File: src/main/java/ru/mail/polis/service/kate/moreva/MySimpleHttpServer.java (hunk 2 above)
Comment (KateMoreva) [reply src/main/java/ru/mail/polis/service/kate/moreva/MySimpleHttpServer.java:111]: Добавила логгер.

**Comment by KateMoreva** [comment]
### Проверка статуса сервера

-t1 -c1 -d30s -R20000 --latency http://localhost:8080/v0/status

**CPU**



Running 30s test @ http://localhost:8080/v0/status
  1 threads and 1 connections
  Thread calibration: mean lat.: 20.280ms, rate sampling interval: 95ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    14.12ms   14.69ms  86.53ms   86.46%
    Req/Sec    20.11k     4.07k   38.01k    71.90%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%    9.74ms
 75.000%   20.80ms
 90.000%   34.01ms
 99.000%   61.89ms
 99.900%   84.03ms
 99.990%   86.46ms
 99.999%   86.59ms
100.000%   86.59ms
  599979 requests in 30.00s, 35.48MB read
Requests/sec:  19999.58
Transfer/sec:      1.18MB

Обработка http запроса 84.29%
Запись в сокет 63.66%
Чтение из сокета 12.29%
Stdev получилось большим, скорее всего, из-за сборок мусора (1%, занимающий более 62ms).

**Alloc**



Running 30s test @ http://localhost:8080/v0/status
  1 threads and 1 connections
  Thread calibration: mean lat.: 13.316ms, rate sampling interval: 63ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    14.45ms   15.36ms  95.42ms   86.30%
    Req/Sec    20.16k     5.06k   39.15k    65.93%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%   10.19ms
 75.000%   20.32ms
 90.000%   35.52ms
 99.000%   68.93ms
 99.900%   92.86ms
 99.990%   95.36ms
 99.999%   95.49ms
100.000%   95.49ms
  599508 requests in 30.00s, 35.45MB read
Requests/sec:  19983.90
Transfer/sec:      1.18MB

Http буфер занимает 99.80%
Обработчик 33.20%
Response ok 22.10%
Возможно, стоит использовать кэширование для экономии 22 процентов.

**Comment by KateMoreva** [comment]
### Put запросы

-t1 -c1 -d30s -R20000 -s ./wrk/put.lua --latency http://localhost:8080

**CPU**


Running 30s test @ http://localhost:8080
  1 threads and 1 connections
  Thread calibration: mean lat.: 339.973ms, rate sampling interval: 2619ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   182.50ms  427.46ms   1.88s    87.73%
    Req/Sec    20.01k     6.77k   32.67k    71.43%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%   11.34ms
 75.000%   40.29ms
 90.000%  851.46ms
 99.000%    1.78s 
 99.900%    1.87s 
 99.990%    1.88s 
 99.999%    1.88s 
100.000%    1.88s 
  598946 requests in 30.00s, 38.27MB read
Requests/sec:  19965.18
Transfer/sec:      1.28MB

Обработка http запроса 72.08%
Запись в сокет 29.94%
Чтение из сокета 9.55%
DAO upsert 26.82%
DAO flush 22.05%

**Alloc**


Running 30s test @ http://localhost:8080
  1 threads and 1 connections
  Thread calibration: mean lat.: 36.802ms, rate sampling interval: 186ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   212.85ms  465.62ms   1.98s    87.17%
    Req/Sec    20.10k     9.39k   43.02k    80.37%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%   23.28ms
 75.000%   60.26ms
 90.000%  974.34ms
 99.000%    1.90s 
 99.900%    1.97s 
 99.990%    1.98s 
 99.999%    1.98s 
100.000%    1.98s 
  599153 requests in 30.00s, 38.28MB read
Requests/sec:  19972.01
Transfer/sec:      1.28MB

Http буфер занимает 99.87%
Обработчик 53.37%
Response ok 7.54%
DAO.upsert 16.06%

Flush не асинхронный, из-за блокировки запросов перед записью на диск возрастает время отклика, от сюда плохое stdev.

**Comment by KateMoreva** [comment]
### Get запросы

-t1 -c1 -d30s -R20000 -s ./wrk/get.lua --latency http://localhost:8080

**CPU**


Running 30s test @ http://localhost:8080
  1 threads and 1 connections
  Thread calibration: mean lat.: 3617.591ms, rate sampling interval: 12886ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    14.25s     4.11s   21.32s    57.53%
    Req/Sec     5.78k     0.00     5.78k     0.00%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%   14.25s 
 75.000%   17.83s 
 90.000%   19.92s 
 99.000%   21.20s 
 99.900%   21.32s 
 99.990%   21.33s 
 99.999%   21.33s 
100.000%   21.33s 
  173574 requests in 30.00s, 12.13MB read
Requests/sec:   5785.87
Transfer/sec:    414.18KB

Обработка http запроса 91.11%
Запись в сокет 4.03%
Чтение из сокета 0.97%
Поиск DAO 85.75%
Поиск ключа в SSTable 67.75% 

**Alloc**


Running 30s test @ http://localhost:8080
  1 threads and 1 connections
  Thread calibration: mean lat.: 3670.728ms, rate sampling interval: 12910ms
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    14.17s     4.07s   21.35s    57.77%
    Req/Sec     5.84k     0.00     5.84k     0.00%
  Latency Distribution (HdrHistogram - Recorded Latency)
 50.000%   14.16s 
 75.000%   17.71s 
 90.000%   19.87s 
 99.000%   21.14s 
 99.900%   21.35s 
 99.990%   21.36s 
 99.999%   21.36s 
100.000%   21.36s 
  172772 requests in 30.00s, 12.08MB read
Requests/sec:   5759.13
Transfer/sec:    412.24KB

Http буфер занимает 99.58%
Обработчик 97.03%
Response 0.86%
Поиск ключа в SSTable 58.01%

**Review by vaddya — APPROVED** [review]
> 90.000% 19.92s

Такой большой latency явно говорит, что это не рабочая нагрузка для сервера. Ставьте в следующий раз поменьше, чтобы в 1-2 секунды укладываться.

Принято

**Comment by KateMoreva** [comment]
> > 90.000% 19.92s
> 
> Такой большой latency явно говорит, что это не рабочая нагрузка для сервера. Ставьте в следующий раз поменьше, чтобы в 1-2 секунды укладываться.
> 
> Принято

Спасибо.

**Pull request closed without merging**